### PR TITLE
SAK-52057 reset-pass migrate to JPA Spring Repository from generic-dao

### PR DESCRIPTION
**Pull request by ern**
https://sakaiproject.atlassian.net/browse/SAK-52057

## Summary by CodeRabbit

- New Features
  - Reworked account validation: new public API, service, repository and DTOs; scheduled background jobs for validation checks; email templates for validation flows.

- Refactor
  - Migration from legacy DAO/HBM to JPA/Spring repositories; controllers and admin flows updated to use the new service API.

- Bug Fixes
  - Improved token expiry, resend behavior, account merge and edge-case handling.

- Tests
  - Added comprehensive unit tests and test wiring for validation scenarios.

- Chores
  - Standardized project metadata and modernized build layout.

### CODE REVIEW

File: reset-pass/account-validator-api/src/main/java/org/sakaiproject/accountvalidator/api/model/ValidationAccount.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import javax.persistence.Transient;
```
[the remark this answers is not in the document]
Comment (ottenhoff) [reply reset-pass/account-validator-api/src/main/java/org/sakaiproject/accountvalidator/api/model/ValidationAccount.java:32]: no no no, we are still on JDK 17 and Spring 5.3 and Hibernate 5.6 per our master/pom.xml file! we will move to JDK 21 and jakarta namespace soon!

File: reset-pass/account-validator-impl/src/test/java/org/sakaiproject/accountvalidator/test/AccountValidationTestConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+import java.util.UUID;
+
+@Configuration
+@EnableTransactionManagement
+@ImportResource("classpath:/WEB-INF/components.xml")
+@PropertySource("classpath:/hibernate.properties")
+public class AccountValidationTestConfiguration extends SakaiTestConfiguration {
```
[the remark this answers is not in the document]
Comment (ern) [reply reset-pass/account-validator-impl/src/test/java/org/sakaiproject/accountvalidator/test/AccountValidationTestConfiguration.java:27]: check the pom.xml it is picked up, one super important thing to note here is that were pulling in the components.xml that is actually used by spring in a running instance and using that in our tests. I can't stress enough of how important it is to do this in unit tests.

File: reset-pass/account-validator-impl/pom.xml [hunk 3]
```diff
@@ -1,162 +1,125 @@
...
+            <groupId>com.sun.mail</groupId>
+            <artifactId>jakarta.mail</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-orm</artifactId>
```
[the remark this answers is not in the document]
Comment (ern) [reply reset-pass/account-validator-impl/pom.xml:42]: we can't upgrade to spring 6 yet

[the remark this answers is not in the document]
Comment (ern) [reply reset-pass/account-validator-impl/pom.xml:42]: also were not using @EnableMethodSecurity meaning were not affected by that vulnerability

File: reset-pass/account-validator-api/src/main/java/org/sakaiproject/accountvalidator/model/ValidationAccount.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        EID                 varchar(255) null,
+        VALIDATION_TOKEN    varchar(255) not null,
+        VALIDATION_SENT     datetime(6)  null,
+        VALIDATION_RECEIVED datetime(6)  null,
+        VALIDATIONS_SENT    int          null,
+        STATUS              int          null,
+        FIRST_NAME          varchar(255) null,
+        SURNAME             varchar(255) null,
+        ACCOUNT_STATUS      int          null
+    ); */
```
[the remark this answers is not in the document]
Comment (ern) [reply reset-pass/account-validator-api/src/main/java/org/sakaiproject/accountvalidator/model/ValidationAccount.java]: yes I will remove, notice the TODO!

File: reset-pass/account-validator-impl/src/test/java/org/sakaiproject/accountvalidator/test/AccountValidationTestConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Bean(name="org.sakaiproject.user.api.PreferencesService")
+    public PreferencesService preferencesService() {
+        return Mockito.mock(PreferencesService.class);
+    }
+
+    @Bean
+    public ResourceLoader resourceLoader() {
+        return Mockito.mock(ResourceLoader.class);
+    }
```
[the remark this answers is not in the document]
Comment (ern) [reply reset-pass/account-validator-impl/src/test/java/org/sakaiproject/accountvalidator/test/AccountValidationTestConfiguration.java:72]: unless the rule is common to more than one test class it makes sense to keep the rule in the test class as it is part of the test logic.

File: reset-pass/account-validator-impl/src/main/java/org/sakaiproject/accountvalidator/impl/service/AccountValidationServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,485 @@
...
+            return false;
+        }
+
+        if (isTokenExpired(va)) {
+            return false;
+        }
+
+        if (va.getValidationReceived() == null) {
+            if (va.getValidationSent() != null) {
+                Instant validationDeadline = Instant.now().minus(VALIDATION_PERIOD_MONTHS, ChronoUnit.DAYS); // 36 months ago
```
Comment (ottenhoff) [inline reset-pass/account-validator-impl/src/main/java/org/sakaiproject/accountvalidator/impl/service/AccountValidationServiceImpl.java:143]: this is kinda strange mix of months and days!
